### PR TITLE
[stable/prometheus-operator] Fix missing rbac.create condition

### PR DESCRIPTION
**Pull request by jkrcma**
#### What this PR does / why we need it:
It's impossible to install the Prometheus subchart into a restricted k8s with `rbac.enable: false` due the forgotten condition.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Review by paskal — APPROVED** [review]

**Comment by vsliouniaev** [comment]
I don't see `rbac.enable` as an option in the values file in this chart

File: stable/prometheus-operator/templates/alertmanager/alertmanager.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ spec:
 {{- end }}
   replicas: {{ .Values.alertmanager.alertmanagerSpec.replicas }}
   listenLocal: {{ .Values.alertmanager.alertmanagerSpec.listenLocal }}
-{{- if .Values.global.rbac.create }}
+{{- if .Values.alertmanager.serviceAccount.name }}
```
Comment (gianrubio) [inline stable/prometheus-operator/templates/alertmanager/alertmanager.yaml]: this is not right, the serviceAccount [will be created](https://github.com/helm/charts/blob/0c3053be740c5d9c3cb2446471ad92aac9bbf217/stable/prometheus-operator/templates/_helpers.tpl#L57-L64) even if `serviceAccount.name` is empty

File: stable/prometheus-operator/templates/prometheus/prometheus.yaml [hunk 2]
```diff
@@ -64,7 +64,7 @@ spec:
   configMaps:
 {{ toYaml .Values.prometheus.prometheusSpec.configMaps | indent 4 }}
 {{- end }}
-{{- if .Values.global.rbac.create }}
+{{- if .Values.prometheus.serviceAccount.name }}
```
Comment (gianrubio) [inline stable/prometheus-operator/templates/prometheus/prometheus.yaml]: same here

**Review by gianrubio — CHANGES_REQUESTED** [review]
please see my previous comments

**Comment by jkrcma** [comment]
@vsliouniaev You are right, stupid mistake from confusion with other charts, was supposed to be `create` instead.

File: stable/prometheus-operator/templates/alertmanager/alertmanager.yaml (hunk 1 above)
Comment (jkrcma) [reply stable/prometheus-operator/templates/alertmanager/alertmanager.yaml]: I see, it breaks the defaults, missed that. Maybe I could omit the whole condition completely?

My use case is to pass the desired SA prometheus/alertmanager should use, created externally.

Comment (jkrcma) [reply stable/prometheus-operator/templates/alertmanager/alertmanager.yaml]: Reworked the approach a bit, can't test it properly so I count on your reviews. Thanks!

File: stable/prometheus-operator/templates/prometheus/prometheus.yaml [hunk 3]
```diff
@@ -64,9 +64,7 @@ spec:
   configMaps:
 {{ toYaml .Values.prometheus.prometheusSpec.configMaps | indent 4 }}
 {{- end }}
-{{- if .Values.global.rbac.create }}
```
Comment (gianrubio) [inline stable/prometheus-operator/templates/prometheus/prometheus.yaml:67]: why did you remove this condition?  It could break the chart when rbac is false and serviceAccount does not exist (it'll not be created by the chart, see here https://github.com/helm/charts/blob/d2e406c9958c6f2275a0243848c0ed0bf2e6bb92/stable/prometheus-operator/templates/prometheus/serviceaccount.yaml

Comment (jkrcma) [reply stable/prometheus-operator/templates/prometheus/prometheus.yaml:67]: When rbac is `false` then `default` account, which should afaik exist at every circumstances (even when RBAC is turned off via k8s flag), is used. https://github.com/helm/charts/blob/d2e406c9958c6f2275a0243848c0ed0bf2e6bb92/stable/prometheus-operator/templates/_helpers.tpl#L62

**Comment by paskal** [comment]
You'll need to bump chart version once more.

**Review by gianrubio — COMMENTED** [review]
/lgtm /ok-to-test

**Comment by paskal** [comment]
@jkrcma you need to merge upstream in order to get merged.
